doc/reference/std/net/websocket.md

```diff
@@ -1 +1,166 @@
 # WebSockets
+
+The `:std/net/websocket` package provides functionality for writing
+websocket clients and servers.
+
+::: tip To use bindings from this module
+```scheme
+(import :std/net/websocket)
+```
+:::
+
+## WebSocket interface
+```scheme
+(defstruct message (data type partial?) final: #t)
+
+(interface (WebSocket Socket)
+  (send (msg :~ valid-message?))
+  (recv) ; -> message
+  (protocol)
+  (max-frame-size))
+
+(valid-message? msg) :-
```

Comment:
Shouldn't this use `:~`?

Reply:
thats my attempt to define it, i guess iy should just be =